packages/@tailwindcss-upgrade/src/utils/packages.ts

```diff
@@ -1,25 +1,127 @@
-import { execSync } from 'node:child_process'
+import { exec as execCb } from 'node:child_process'
 import fs from 'node:fs/promises'
 import { dirname, resolve } from 'node:path'
+import { promisify } from 'node:util'
+import { DefaultMap } from '../../../tailwindcss/src/utils/default-map'
 import { warn } from './renderer'
 
-let didWarnAboutPackageManager = false
+const exec = promisify(execCb)
+
+export function pkg(base: string) {
+  return {
+    async add(packages: string[], location: 'dependencies' | 'devDependencies' = 'dependencies') {
+      let packageManager = await packageManagerForBase.get(base)
+      return packageManager.add(packages, location)
+    },
+    async remove(packages: string[]) {
+      let packageManager = await packageManagerForBase.get(base)
+      return packageManager.remove(packages)
+    },
+  }
+}
+
+class PackageManager {
+  constructor(private base: string) {}
+
+  async exec(command: string) {
+    return exec(command, { cwd: this.base })
+  }
+
+  async add(
+    packages: string[],
+    location: 'dependencies' | 'devDependencies',
+  ): ReturnType<typeof this.exec> {
+    throw new Error('Method not implemented.')
+  }
+
+  async remove(packages: string[]): ReturnType<typeof this.exec> {
+    throw new Error('Method not implemented.')
+  }
+}
```

Comment:
Could we simplify all this subclassing stuff by just relying on the fact that `{packageManager} add {library} -D` works to save a dev dependency in npm, pnpm, yarn, and bun?

Then it's more like the old implementation which was a lot less code.

Reply:
Guess Bun doesn't support `-D`, sad.

Reply:
Yep was looking at the documentation for each and there are slight differences. But can still make it smaller and just encode the exceptions. Let me adjust.

Reply:
@adamwathan simplified it by just encoding the exception right now: https://github.com/tailwindlabs/tailwindcss/pull/14830/commits/a8f54dd1597326876579bdb61e07d0fcba02895b